pyhf/tensor/tensorflow_backend.py

```diff
@@ -44,8 +44,11 @@ def clip(self, tensor_in, min, max):
         return tf.clip_by_value(tensor_in, min, max)
 
     def tolist(self,tensor_in):
-        tensor_in = self.astensor(tensor_in)
-        return self.session.run(tensor_in).tolist()
+        try:
+            return self.session.run(tensor_in).tolist()
+        except AttributeError:
```

Comment:
if there is no session we should raise with a message that tells us what's going on

```
if not self.session: raise RuntimeError('evaluation of tensor requested via .tolist() but no session defined')
```

or sth